### PR TITLE
Fix assertions compilation/rendering when extensions are involved

### LINKED ISSUE #50: Implicit conversion problem on Scala 3
Here's a scastie: https://scastie.scala-lang.org/oS3IvTm8RLCaFxII64wkbw

Here's the code:

```scala
import com.eed3si9n.expecty.Expecty.expect

trait Eq[T] {
  def eq(x: T, y: T): Boolean
}

object Eq {
  implicit val eqInt: Eq[Int] = new Eq[Int] {
    def eq(x: Int, y: Int) = (x - y) == 0
  }
}

implicit class EqOps[T](i: T)(implicit eq: Eq[T]) { 
  def `===`(other: T) = eq.eq(i, other)
}

@main def hello = 
  expect(1 === 1)
```

And it fails with

```
Found:    main$package.EqOps[Int]
Required: (implicit eq: Eq[Int]): main$package.EqOps[Int]
```

According to https://github.com/disneystreaming/weaver-test/issues/313 it behaves differently on 2.13

### PR DESCRIPTION
**Pull request by Baccata**
Fixes https://github.com/eed3si9n/expecty/issues/50 

From what I gather, in the case of extension methods, the sub-type-tree that the macro is trying to capture to pass to the recorder doesn't quite match what the recorder expects. 

This PR adds logic to hopefully detect implicit extensions and avoid capture the extension instance.

### CODE REVIEW

**Comment by keynmol** [comment]
We're just debugging the output that happens with this fix:

```scala
[info] - extensionMethods *** FAILED ***
[info]   assertion failed
[info]
[info]   2 === 1
[info]   |||
[info]   ||false
[info]   |foo.ExpectyTest$$anon$1@5d995b92
[info]   foo.ExpectyTest$EqOps$1@43786fe9
[info]     com.eed3si9n.expecty.ExpectyBase$ExpectyListener.expressionRecorded(Expecty.scala:41)
[info]     com.eed3si9n.expecty.RecorderRuntime.recordExpression(RecorderRuntime.scala:40)
[info]     foo.ExpectyTest$.$init$$$anonfun$5(ExpectyTest.scala:66)
[info]     verify.BasicTestSuite.test$$anonfun$1(BasicTestSuite.scala:23)
[info]     verify.TestSpec$.liftedTree2$1(TestSpec.scala:48)
[info]     verify.TestSpec$.sync$$anonfun$1(TestSpec.scala:57)
[info]     verify.TestSpec.apply(TestSpec.scala:22)
[info]     verify.Properties.liftedTree1$1(Properties.scala:34)
[info]     verify.Properties.liftedTree2$3(Properties.scala:38)
[info]     verify.Properties.iterator$$anonfun$2$$anonfun$1(Properties.scala:50)
[info]     verify.TestSpec.apply(TestSpec.scala:22)
[info]     verify.runner.Task.loop$2(Task.scala:51)
[info]     verify.runner.Task.loop$1$$anonfun$1(Task.scala:58)
[info]     scala.concurrent.impl.Promise$Transformation.run(Promise.scala:434)
[info]     java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
[info]     java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[info]     java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[info]     java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[info]     java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[info]     java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Comment by eed3si9n** [comment]
Thanks guys! Could you kindly add a rendering test for the failed test please?
If the rendering contains `foo.ExpectyTest$EqOps$1@43786fe9` maybe we should open another issue for that, and try to improve it later?

**Comment by Baccata** [comment]
> maybe we should open another issue for that, and try to improve it later?

If you don't mind waiting a little (maybe a couple days at most), I'm getting closer to addressing rendering and might be able to ship everything at once.

**Comment by eed3si9n** [comment]
Sounds good to me.

**Comment by Baccata** [comment]
Rendering is "fixed", ie doesn't contain `foo.ExpectyTest$EqOps$1@43786fe9` anymore. Could do with some sanity check @keynmol

**Comment by eed3si9n** [comment]
Nice work @Baccata!

**Review by keynmol — APPROVED** [review]
Awesome!

**Comment by keynmol** [comment]
Tested this with the Scala 3 extension methods as well - seems to work fine

**Comment by keynmol** [comment]
I also published it locally and ran some weaver tests with it - 2.13 and 3.0 seem fine

**Comment by Baccata** [comment]
Thanks ! @eed3si9n waiting for your blessing to merge

**Review by eed3si9n — APPROVED** [review]
Overall LGTM
If it's not too much work, could you squash the commits so it might be easier to bisect / cherry-pick later?

**Comment by Baccata** [comment]
Squashed into 0406cd2 :)

**Comment by keynmol** [comment]
@Baccata @eed3si9n if there's no objections, I will release it as **0.15.4** tomorrow

I want to get it into weaver ASAP, as I think original case was underreported - I'm hitting this issue even just with straight methods that take implicit parameters, so it's pretty bad :)

**Pull request merged**
